compendium/DeclarativeServices/src/SCRAsyncWorkService.cpp

```diff
@@ -50,22 +55,23 @@ class SCRAsyncWorkServiceDetail
 
   void Shutdown()
   {
-    try {
-      if (threadpool) {
-        try {
-          threadpool->join();
-          threadpool->stop();
-          threadpool.reset();
-        } catch (...) {
-          //
-        }
+    if (threadpool) {
+      try {
+        threadpool->join();
+        threadpool->stop();
+        threadpool.reset();
+      } catch (...) {
+        std::string msg =
+          "An exception has occured while trying to shutdown "
+          "the fallback cppmicroservices::async::detail::AsyncWorkService "
+          "instance.";
+        logger->Log(cppmicroservices::logservice::SeverityLevel::LOG_WARNING,
+                    msg);
```

Comment:
I suggest using the `Log` function overload that takes a `std::exception_ptr` and capturing the exception.